### ISSUE TITLE
Unable to deploy to CF Pages

### ISSUE THREAD

**Issue by cbarratt** [storyblok/storyblok-astro#46]
I'm trying to use this package and deploy to CFlare pages, both locally when doing `astro build && wrangler pages dev ./dist`

I'm seeing the following error:

```
storyblokApiInstance has not been initialized correctly
14:12:25 [ssr] TypeError: Cannot read properties of undefined (reading 'get')
```

I can't introspect the CF pages logs to see this, but im also seeing a 500 error on my page (the same as locally), so presumably its having the same issue.

Any advice would be appreciated

**Comment by manuelschroederdev** [comment]
Hi @cbarratt, does it build fine locally?

**Comment by cbarratt** [comment]
@manuelschroederdev - it also fails in the same manner locally too when doing `astro build && wrangler pages dev ./dist`

**Comment by manuelschroederdev** [comment]
Ah I see, thanks for letting me know. I'll have a look asap.

**Comment by cbarratt** [comment]
@manuelschroederdev - no worries, I've just decided to try hosting it on a standard node runtime with Astro running with this config:

```
  output: "server",
  server: { host: true },
  adapter: node({
    mode: 'standalone'
  }),
```

And it builds fine with no errors, but booting the app on fly.io host is giving the same feedback:

```
2022-10-20T13:38:40Z app[1278d0b1] lhr [info]storyblokApiInstance has not been initialized correctly
2022-10-20T13:38:40Z app[1278d0b1] lhr [info]01:38:40 PM [ssr] TypeError: Cannot read properties of undefined (reading 'get')
```

**Comment by Soviut** [comment]
I get the same error when I try to use SSR `output: 'server'` on CloudFlare. The worker debug logs look like the following.

```json
{
      "message": [
        "storyblokApiInstance has not been initialized correctly"
      ],
      "level": "error",
      "timestamp": 1671719795829
    },
    {
      "message": [
        "02:36:35 PM [ssr] TypeError: Cannot read properties of undefined (reading 'get')\n    at worker.mjs:575:1568\n    at Ku (worker.mjs:63:645)\n    at hl (worker.mjs:85:2966)\n    at async hr.fr (worker.mjs:85:8956)\n    at async hr.render (worker.mjs:85:8123)\n    at async Object.fetch (worker.mjs:85:10517)"
      ],
      "level": "error",
      "timestamp": 1671719795829
    },
```

That error message

```
02:36:35 PM [ssr] TypeError: Cannot read properties of undefined (reading 'get')
    at worker.mjs:575:1568
    at Ku (worker.mjs:63:645)
    at hl (worker.mjs:85:2966)
    at async hr.fr (worker.mjs:85:8956)
    at async hr.render (worker.mjs:85:8123)
    at async Object.fetch (worker.mjs:85:10517)
```

As far as I can tell, the CloudFlare deployment works fine.

**Comment by jaydanurwin** [comment]
@manuelschroederdev First of all thank you so much for your work on this repo to point me in the right direction for my `astro-sanity` package linked above.

Have you made any progress on SSR support? To me this seems like either an Astro or a Vite limitation. I've tried a virtual plugin (as it seems you have too) and have not had any luck. 

`globalThis` seems to work fine in static mode but in SSR it will run in the dev server and build successfully but can't serve. Do we need to reach out to the Astro team? It seems like pretty table-stakes feature to be able to pass a config object down into an integration then the integration makes that config globally available.

**Comment by manuelschroederdev** [comment]
Hey @cbarratt and @Soviut, unfortunately, I haven't been able to test this yet - is this still a problem with CloudFare pages and the latest versions of Astro and Storyblok integration?

**Comment by Soviut** [comment]
I found a solution but forgot about this ticket.

The issue was that Storyblok requires SSL to use their visual editor, even with localhost. So I installed [vite-plugin-basic-ssl](https://github.com/vitejs/vite-plugin-basic-ssl).

In my astro.config.mjs
```js
vite: {
  server: { https: true },
  plugins: [basicSsl()],
}
```

It turns out that was trying to run inside the CloudFlare worker and triggering the aforementioned errors. So I needed to only load the plugin while local so I did the following:

```js
vite: {
  server: { https: import.meta.env.DEV },
  plugins: import.meta.env.DEV ? [basicSsl()] : [],
}
```
